solv/simplequery.c

```diff
@@ -0,0 +1,134 @@
+/*
+ * Copyright (C) 2022 VMware, Inc. All Rights Reserved.
+ *
+ * Licensed under the GNU Lesser General Public License v2.1 (the "License");
+ * you may not use this file except in compliance with the License. The terms
+ * of the License are located in the COPYING file of this distribution.
+ */
+
+#include "includes.h"
+
+#define ASSERT_ARG(x) { \
+    if (!(x)) { \
+        dwError = ERROR_TDNF_INVALID_PARAMETER; \
+        BAIL_ON_TDNF_LIBSOLV_ERROR(dwError); \
+    } \
+}
+
+#define ASSERT_MEM(x) { \
+    if (!(x)) { \
+        dwError = ERROR_TDNF_OUT_OF_MEMORY; \
+        BAIL_ON_TDNF_LIBSOLV_ERROR(dwError); \
+    } \
+}
+
+/* Split nevra into name, ever, arch. This modifies its input,
+   and returns pointers to the respective components */
+static
+int split_nevra(char *nevra, char **name, char **evr, char **arch)
+{
+    char *p = nevra + strlen(nevra) - 1;
+
+    while (*p != '.' && p > nevra) p--;
+    if (p <= nevra) {
+        return -1;
+    }
+    *p = 0;
+    *arch = p+1;
+    p--;
+
+    while (*p != '-' && p > nevra) p--;
+    if (p <= nevra){
+        return -2;
+    }
+    p--;
+    while (*p != '-' && p > nevra) p--;
+    if (p <= nevra) {
+        return -3;
+    }
+    *p = 0;
+    *evr = p+1;
+
+    *name = nevra;
+
+    return 0;
+}
+
+/* Find packages by nevra as specified with ids. Must be either installed
+   or not as set by the 'installed' flag. Adds result to qresult, can
+   be multiples if package is in multiple repos. */
+uint32_t
+SolvFindSolvablesByNevraId(
+    Pool *pool,
+    Id name,
+    Id evr,
+    Id arch,
+    Queue* qresult,
+    int installed
+    )
+{
+    uint32_t dwError = 0;
+
+    ASSERT_ARG(pool);
+    ASSERT_ARG(qresult);
+
+    Id p;
```

Comment:
Isn't p should be initialised? Below get used as index

Reply:
No need, see https://github.com/openSUSE/libsolv/blob/6ffe49cdd5c4f53b9135a51179e3b7c160a472aa/src/pool.h#L409